src/internal/easee/controller.go

```diff
@@ -97,3 +113,64 @@ func (c *controller) ElectricityMeterReport(unit string) (float64, error) {
 		return 0, errors.Errorf("unsupported unit: %s", unit)
 	}
 }
+
+func (c *controller) SetChargepointChargingMode(mode string) error {
+	switch mode {
+	case ChargingModeSlow:
+		return c.client.SetChargingCurrent(c.chargerID, SlowChargingCurrentInAmpers)
+	case ChargingModeNormal:
+		return c.client.SetChargingCurrent(c.chargerID, NormalChargingCurrentInAmpers)
+	default:
+		return errors.Errorf("unsupported mode: %s", mode)
+	}
+}
+
+func (c *controller) ChargepointChargingModeReport() (string, error) {
+	state, err := c.client.ChargerState(c.chargerID)
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch charger state for ID: %s: %w", c.chargerID, err)
+	}
+
+	if state.ChargerOpMode.String() != ChargerModeCharging {
+		return "", fmt.Errorf("cannot report charging mode if device is not charging")
+	}
+
+	current := state.DynamicChargerCurrent
+	if current < 0 {
+		return "", fmt.Errorf("cannot assign charging mode: reported current is lower than zero")
+	}
+
+	if current > SlowChargingCurrentInAmpers {
```

Comment:
IMHO, this method should just return what is set in the persistent configuration of the device stored by the adapter. It's not about the current charging speed.

Alterantively, if it's not charging it should return persisted configuration, and if it is charging, then it should report what is actually set on the device? Not sure about it.

Reply:
Outdated.